packages/wallet-adapter-core/src/WalletCore.ts

```diff
@@ -564,6 +568,91 @@ export class WalletCore extends EventEmitter<WalletCoreEvents> {
     }
   }
 
+  async signIn(
+    input: AptosSignInInput,
+    walletName: string
+  ): Promise<void | AptosSignInOutput> {
+    const allDetectedWallets = this._standard_wallets;
+    const selectedWallet = allDetectedWallets.find(
+      (wallet: AdapterWallet) => wallet.name === walletName
+    );
+
+    if (!selectedWallet) {
+      throw new WalletNotFoundError(`Wallet ${walletName} not found`).message;
+    }
+
+    if (!selectedWallet.features["aptos:signIn"]) {
+      throw new WalletNotSupportedMethod(
+        `aptos:signIn is not supported by ${walletName}`
+      ).message;
+    }
+
+    // Check if we are in a redirectable view (i.e on mobile AND not in an in-app browser)
+    // If we are, navigate to the mobile app's browser instead.
+    //
+    // Ignore if wallet is installed (iOS extension)
+    if (isRedirectable()) {
+      if (selectedWallet.readyState === WalletReadyState.Installed) {
+        // If wallet has a openInMobileApp method, use it
+        if (selectedWallet.features["aptos:openInMobileApp"]?.openInMobileApp) {
+          selectedWallet.features["aptos:openInMobileApp"]?.openInMobileApp();
+          return;
+        }
+      }
+
+      if (selectedWallet.readyState === WalletReadyState.NotDetected) {
+        // If wallet has a deeplinkProvider property, i.e wallet is on the internal registry, use it
+        const uninstalledWallet =
+          selectedWallet as unknown as AptosStandardSupportedWallet;
+        if (uninstalledWallet.deeplinkProvider) {
+          const url = encodeURIComponent(window.location.href);
+          const location = uninstalledWallet.deeplinkProvider.concat(url);
+          window.location.href = location;
+          return;
+        }
+      }
+    }
+
+    return this.signInWallet(input, selectedWallet);
+  }
+
+  async signInWallet(
+    input: AptosSignInInput,
+    selectedWallet: AdapterWallet
+  ): Promise<void | AptosSignInOutput> {
```

Comment:
dont think we need `void`
```suggestion
  ): Promise<AptosSignInOutput> {
```

Reply:
The issue is that when we redirect to the mobile app, we can't really return anything since there's no output. I could possibly remove the redirect behavior since the flow should be...

1. Attempt to `signIn`
2. If not supported, attempt to regular `connect`
3. Regular `connect` will redirect to mobile app

Reply:
Removed from the flow